include/drm/drm_gem.h

```diff
@@ -119,6 +119,11 @@ struct drm_gem_object {
 	 * simply leave it as NULL.
 	 */
 	struct dma_buf_attachment *import_attach;
+
+	/*
+	 * is backed by scattered pages
+	 */
+	bool is_scattered;
```

Comment:
Could you use a common struct base for meson GEM, and then embed scattered / CMA in a union?